android/KMAPro/kMAPro/src/main/AndroidManifest.xml

```diff
@@ -193,25 +193,39 @@
       </intent-filter>
 
       <intent-filter android:autoVerify="true">
```

Comment:
I think for clarity there should be a separate `intent-filter` for each URI. As it stands, I'm pretty sure it's not quite right.

There are four endpoints:

https://keyman-staging.com/go/package/download
https://keyman.com/go/package/download
https://keyman-staging.com/keyboards/install
https://keyman.com/keyboards/install

```xml
<intent-filter android:autoVerify="true">
  <action android:name="android.intent.action.VIEW" />
  <category android:name="android.intent.category.DEFAULT" />
  <category android:name="android.intent.category.BROWSABLE" />
  <data android:scheme="https" android:host="keyman-staging.com" android:pathPrefix="/go/package/download/" />
  <data android:scheme="http" />
</intent-filter>
<intent-filter android:autoVerify="true">
  <action android:name="android.intent.action.VIEW" />
  <category android:name="android.intent.category.DEFAULT" />
  <category android:name="android.intent.category.BROWSABLE" />
  <data android:scheme="https" android:host="keyman.com" android:pathPrefix="/go/package/download/" />
  <data android:scheme="http" />
</intent-filter>
<intent-filter android:autoVerify="true">
  <action android:name="android.intent.action.VIEW" />
  <category android:name="android.intent.category.DEFAULT" />
  <category android:name="android.intent.category.BROWSABLE" />
  <data android:scheme="https" android:host="keyman-staging.com" android:pathPrefix="/keyboards/install/" />
  <data android:scheme="http" />
</intent-filter>
<intent-filter android:autoVerify="true">
  <action android:name="android.intent.action.VIEW" />
  <category android:name="android.intent.category.DEFAULT" />
  <category android:name="android.intent.category.BROWSABLE" />
  <data android:scheme="https" android:host="keyman.com" android:pathPrefix="/keyboards/install/" />
  <data android:scheme="http" />
</intent-filter>
```

Reply:
k, I've split it to two intent-filters to handle the 4 endpoints

Reply:
I still don't think the intent filters are right. Each one has to have both `http` and `https` for *both* `keyman.com` and `keyman-staging.com`

Reply:
I was going by the https://developer.android.com/training/app-links/deep-linking#adding-filters documentation

>  ...because multiple `<data>` elements in the same intent filter are actually merged together to account for all variations of their combined attributes. For example, consider the following:
```
<intent-filter>
  ...
  <data android:scheme="https" android:host="www.example.com" />
  <data android:scheme="app" android:host="open.my.app" />
</intent-filter>
```
>
>It might seem as though this supports only `https://www.example.com` and `app://open.my.app`. However, it actually supports those two, plus these: `app://www.example.com` and `https://open.my.app`.

I tried to summarize that in the comment l.198

Reply:
Ah, gotcha. That's ... less than obvious! I read that documentation several times and missed that.